python/ray/data/_internal/execution/operators/input_data_buffer.py

```diff
@@ -0,0 +1,43 @@
+from typing import List, Optional
+
+from ray.data._internal.stats import StatsDict
+from ray.data._internal.execution.interfaces import (
+    RefBundle,
+    PhysicalOperator,
+)
+
+
+class InputDataBuffer(PhysicalOperator):
+    """Defines the input data for the operator DAG.
+
+    For example, this may hold cached blocks from a previous Dataset execution, or
+    the arguments for read tasks.
+    """
+
+    def __init__(self, input_data: List[RefBundle]):
```

Comment:
Hmm I thought the inputs to read tasks wouldn't always be Refs, like a list of filenames?

Reply:
Yeah, that would be true once we add the Read operator (which doesn't need to subclass from InputDataBuffer). Currently, in the legacy translation layer we ray.put() the read tasks, but this should go away in the future.